File: protocol-generator/java/src/test/java/org/entitycore/protocol/conformance/ConformanceTest.java
package org.entitycore.protocol.conformance;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

/** Drives the ECF wire-conformance corpus and asserts 0 FAIL (the S2 phase gate). */
final class ConformanceTest {

    @Test
    void ecfCorpusByteIdentical() throws Exception {
        ConformanceHarness.Result r = ConformanceHarness.run(ConformanceHarness.defaultFixture());
        System.out.printf("== ECF conformance: %d/%d PASS, %d FAIL ==%n",
                r.pass(), r.total(), r.fail());
        for (String f : r.failures()) {
            System.out.println("  " + f);
        }
        assertTrue(r.total() >= 69, "expected at least 69 testable vectors, got " + r.total());
        assertEquals(0, r.fail(), "wire-conformance must be all-PASS");
    }
}

File: protocol-generator/kotlin/src/test/kotlin/org/entitycore/protocol/conformance/ConformanceTest.kt
package org.entitycore.protocol.conformance

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

/**
 * The ECF conformance gate as a kotlin.test test (JUnit 5 platform). Decodes the
 * vendored fixture with our own decoder, runs every vector, byte-compares against the
 * embedded cross-blessed `canonical` bytes. All vectors MUST pass.
 */
class ConformanceTest {
    @Test
    fun allVectorsByteIdentical() {
        val result = ConformanceHarness.run(ConformanceHarness.defaultFixture())
        println("ECF conformance: ${result.pass}/${result.total} PASS (${result.fail} fail)")
        result.failures.forEach { println("  $it") }
        assertTrue(result.total >= 69, "expected >= 69 testable vectors, got ${result.total}")
        assertEquals(0, result.fail, "byte-identity failures:\n" + result.failures.joinToString("\n"))
        assertEquals(result.total, result.pass)
    }
}
